Lua/Indent.tmPreferences

```diff
@@ -4,13 +4,13 @@
 	<key>name</key>
 	<string>Indent</string>
 	<key>scope</key>
-	<string>source.lua</string>
+	<string>source.lua - string - comment</string>
 	<key>settings</key>
 	<dict>
 		<key>decreaseIndentPattern</key>
-		<string>^\s*(elseif|else|end|\})\s*$</string>
+		<string>(^(?!=.*(then|do).*).*\b(elseif|else|end|until)\b(?!.*(\"|\'|\[\=*\[)).*$|^((?!\{).)*\}\;?.*$)</string>
```

Comment:
it'd be great if you could format these regex patterns over multiple lines (i.e. `(?x)` mode) to make them readable and add some comments explaining what they do